core/network-libp2p/src/config.rs

```diff
@@ -206,7 +208,9 @@ where
 			if e.kind() == io::ErrorKind::NotFound {
 				file.as_ref().parent().map_or(Ok(()), fs::create_dir_all)?;
 				let sk = generate();
-				write_secret_file(file, sk.as_ref())?;
+				let mut sk_vec = serialize(&sk);
+				write_secret_file(file, &sk_vec)?;
+				for b in sk_vec.iter_mut() { *b = 0; }  // Zero the private key
```

Comment:
Since [zeroize](https://crates.io/crates/zeroize) is already a transitive dependency, as it is used in libp2p-core, how about just using it here by importing `zeroize::Zeroize` and then `sk_vec.zeroize()`?